addons/hr_holidays/models/hr_leave_allocation.py

```diff
@@ -258,7 +258,7 @@ def _compute_number_of_days(self):
             allocation_unit = allocation.type_request_unit
             if allocation_unit != 'hour':
                 allocation.number_of_days = allocation.number_of_days_display
-            else:
+            elif allocation.employee_id:
```

Comment:
```suggestion
            elif allocation_unit == 'hour' and allocation.employee_id:
```
I know this might be an overkill but it is for better readability and understanding for anyone who is gonna read this line of code in the future